# Food Hub 

## Context
Restaurant aggregator app that collects local restaurants in one place. It allows customers to place order to their favorite restaurant and have it delivered.
A unique order is created when a customer places and order with a particular restaurant, FoodHub then assigns this delivery to one of its partner driver to go pick up the food from the restaurant an deliver to the customer. 

## Objective
The objective of the analysis is exploratory, to find out general insights around demand in the ecosystem. Is there a cuisine that does better in a particular region? Is there a particular time of day that customers prefer to order at?

In [12]:
# Import Libraries for data manipulation
import pandas as pd
import numpy as np

# Import Libraries for data visualization
import matplotlib.pyplot as plt
import seaborn as sns

In [13]:
pd.set_option('display.float_format', lambda x: '%.5f' % x)

In [14]:
# Read the data
df = pd.read_csv('foodhub_order.csv')
# Return first 5 rows
df.head()

order_id  customer_id            restaurant_name cuisine_type  \
0   1477147       337525                    Hangawi       Korean   
1   1477685       358141  Blue Ribbon Sushi Izakaya     Japanese   
2   1477070        66393                Cafe Habana      Mexican   
3   1477334       106968  Blue Ribbon Fried Chicken     American   
4   1478249        76942           Dirty Bird to Go     American   

   cost_of_the_order day_of_the_week     rating  food_preparation_time  \
0           30.75000         Weekend  Not given                     25   
1           12.08000         Weekend  Not given                     25   
2           12.23000         Weekday          5                     23   
3           29.20000         Weekend          3                     25   
4           11.59000         Weekday          4                     25   

   delivery_time  
0             20  
1             23  
2             28  
3             15  
4             24

##### Observation  
The data has 9 columns. Each row represents a unique order. 
Some customers did not give ratings.

In [16]:
df.shape

(1898, 9)

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1898 entries, 0 to 1897
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               1898 non-null   int64  
 1   customer_id            1898 non-null   int64  
 2   restaurant_name        1898 non-null   object 
 3   cuisine_type           1898 non-null   object 
 4   cost_of_the_order      1898 non-null   float64
 5   day_of_the_week        1898 non-null   object 
 6   rating                 1898 non-null   object 
 7   food_preparation_time  1898 non-null   int64  
 8   delivery_time          1898 non-null   int64  
dtypes: float64(1), int64(4), object(4)
memory usage: 133.6+ KB


* The data set has 1898 rows and 9 columns. 
* The order and customer ID are both of the Int data type. Which is OK
* Restaurant name, cusine type, day of the week are object data type, We are going to change these to category.  
* Ratings however should be numbers / Int  
* Cost of order is of the float data type. Which is OK for money  
* Food prep time and delivery time should be date_time not Int


In [21]:
# Changing the data type will reduce the overall size of the data set. Which will improve performance.
df.restaurant_name = df.restaurant_name.astype('category')
df.cuisine_type = df.cuisine_type.astype('category')
df.day_of_the_week = df.day_of_the_week.astype('category')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1898 entries, 0 to 1897
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype   
---  ------                 --------------  -----   
 0   order_id               1898 non-null   int64   
 1   customer_id            1898 non-null   int64   
 2   restaurant_name        1898 non-null   category
 3   cuisine_type           1898 non-null   category
 4   cost_of_the_order      1898 non-null   float64 
 5   day_of_the_week        1898 non-null   category
 6   rating                 1898 non-null   object  
 7   food_preparation_time  1898 non-null   int64   
 8   delivery_time          1898 non-null   int64   
dtypes: category(3), float64(1), int64(4), object(1)
memory usage: 102.7+ KB


In [22]:
# summary statistics of the dataset
df.describe()

order_id  customer_id  cost_of_the_order  food_preparation_time  \
count    1898.00000   1898.00000         1898.00000             1898.00000   
mean  1477495.50000 171168.47840           16.49885               27.37197   
std       548.04972 113698.13974            7.48381                4.63248   
min   1476547.00000   1311.00000            4.47000               20.00000   
25%   1477021.25000  77787.75000           12.08000               23.00000   
50%   1477495.50000 128600.00000           14.14000               27.00000   
75%   1477969.75000 270525.00000           22.29750               31.00000   
max   1478444.00000 405334.00000           35.41000               35.00000   

       delivery_time  
count     1898.00000  
mean        24.16175  
std          4.97264  
min         15.00000  
25%         20.00000  
50%         25.00000  
75%         28.00000  
max         33.00000

* The average cost of meals is $16.49, ranging between $4.47 to $35.41 
* Majority (75%) of the orders are below $22.29
* The avg time to prepare food is 27 minutes ranging between 20 and 35 minutes. 
* Majority (75%) of the orders take below 31 minutes
* The avg time to deliver food is 24.16 minutes ranging between 15 and 33 minutes
* Majority (75%) of the food deliveries take a about 28 minutes

In [ ]:
# How many orders are not rated?
df.rating.value_counts()

Not given    736
5            588
4            386
3            188
Name: rating, dtype: int64

**There are 736 orders that were not rated.**

# Exploratory Data Analysis

## Univariate Analysis